### ISSUE TITLE
Decomplect xGroupBy

### ISSUE THREAD

**Issue by rgrannell1** [rgrannell1/kea#57]
Kea's implementation of groupBy includes both the output of its partitioning function and the partitions, while Haskell and SQL only return the latter.
## Current:

``` r
xGroupBy(x. %% 2, 1:5)
list(list(1, list(1L, 3L, 5L)), list(0, list(2L, 4L)))
```
## Target:

``` r
xGroupBy(x. %% 2, 1:5)
list(list(1L, 3L, 5L), list(2L, 4L))
```

There needs to be a light replacement for the current functionality of `xGroupBy`: it is very useful to be able to index these partitions by their shared value. This combination of combinators should be clearly documented.

**Comment by rgrannell1** [comment]
xGroupBy could be complemented by a simpler brother, as it is still awkward to regenerate the indices after grouping, but on the other hand you often don't need the indices.
Possible names:
- `xCollectBy`
- `xMusterBy`
- `xAssortBy`
- `xAmassBy`

collect is too vague; assort might be confused for sort; muster is too obscure, so the last name is probably best.

**Comment by rgrannell1** [comment]
Closed in v0.78.0
